src/timeline/RoomlistModel.cpp

```diff
@@ -1105,3 +1123,100 @@ FilteredRoomlistModel::previousRoom()
         }
     }
 }
+
+#ifdef NHEKO_DBUS_SYS
+RoomListDBusInterface::RoomListDBusInterface(RoomlistModel *parent)
+  : QObject{parent}
+  , m_parent{parent}
+{
+    connect(ChatPage::instance(), &ChatPage::newRoom, this, &RoomListDBusInterface::prepareModel);
+    connect(ChatPage::instance(), &ChatPage::leftRoom, this, &RoomListDBusInterface::prepareModel);
+    prepareModel();
+}
+
+QVector<nheko::dbus::RoomInfoItem>
+RoomListDBusInterface::getRooms(const QDBusMessage &message)
+{
+    auto reply = message.createReply();
+
+    m_modelAccess.lock();
+    nhlog::ui()->debug("Sending {} rooms over D-Bus...", m_model->size());
+    reply << QVariant::fromValue(*m_model);
+    QDBusConnection::sessionBus().send(reply);
+    nhlog::ui()->debug("Rooms successfully sent to D-Bus.");
+    m_modelAccess.unlock();
+
+    return {};
+}
+
+void
+RoomListDBusInterface::activateRoom(const QString &roomid) const
+{
+    bringWindowToTop();
+    m_parent->setCurrentRoom(roomid);
+}
+
+void
+RoomListDBusInterface::joinRoom(const QString &roomid) const
+{
+    bringWindowToTop();
+    ChatPage::instance()->joinRoom(roomid);
+}
+
+void
+RoomListDBusInterface::prepareModel()
+{
+    m_modifyStagingDataMutex.lock();
+    static int modelSize;
+    modelSize = m_parent->models.size();
+    if (modelSize == 0) {
+        m_modifyStagingDataMutex.unlock();
+        return;
+    }
+
+    for (const auto &model : std::as_const(m_parent->models)) {
+        MainWindow::instance()->imageProvider()->download(
+          model->roomAvatarUrl().remove("mxc://"),
+          {32, 32},
```

Comment:
This should probably be bigger for other use cases. Maybe 128x128 or 96x96?

Reply:
Or can this be configurable? Maybe it should be a second call? Not sure how to best approach that :3

Reply:
I think just hard-coding 128x128 should do the trick. It would be too complex to add configuration IMO, and if anybody wants that later on, they are welcome to fix it.